tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/NetStandardLibraries/RestSharp/RestSharp107andBeyondExerciser.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using RestSharp;
+using NewRelic.Agent.IntegrationTests.Shared.ReflectionHelpers;
+using NewRelic.Api.Agent;
+using System.Runtime.CompilerServices;
+
+// .NET 4.8 and 4.8.1 test v107+ of RestSharp which has a different API than older versions
+#if NET48_OR_GREATER
+
+namespace MultiFunctionApplicationHelpers.NetStandardLibraries.RestSharp
+{
+    [Library]
+    public class RestSharpExerciser
+    {
+        private class Bird
+        {
+            public string CommonName { get; set; }
+            public string BandingCode { get; set; }
+        }
+
+        [LibraryMethod]
+        [Transaction]
+        [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
+        public string SyncClient(int port, string method, bool generic)
+        {
+            var myHost = "localhost"; // Request.RequestUri.Host;
+            var myPort = port;
+            var client = new RestClient($"http://{myHost}:{myPort}");
+
+            var endpoint = "api/RestAPI/";
+            var id = 1;
+
+            var requests = GetRequests(endpoint, id);
+            if (generic)
+            {
+                // This may cause hangs...
+                var response = client.ExecuteAsync<Bird>(requests[method]).GetAwaiter().GetResult();
+                if (method == "GET")
+                {
+                    var bird = response.Data;
+                    System.IO.File.AppendAllText(@"C:\IntegrationTestWorkingDirectory\RestAPIController.log", $"SyncClient method={method}, generic={generic} got Bird {bird.CommonName} ({bird.BandingCode})" + Environment.NewLine);
```

Comment:
I think I've mentioned this before, but hard-coded paths are generally a bad idea, especially if we want to get these running on Linux one day. This doesn't have to be addressed now, though.

Reply:
Yeah, I copied the existing impl for the `RestSharpExerciser` which does the same thing... Honestly, I don't think we need to write to the filesystem in these test exercisers.. I'll convert this.

Reply:
Awesome, thanks!